geonode/maps/templates/leaflet_maps/map_edit.html

```diff
@@ -473,30 +472,62 @@ <h4 class="modal-title">Map Updated</h4>
                 }
             }); // That's all
 
-            {% for layer in map_layers %}
```

Comment:
Heavy refactor. This is generally not good, because it produce duplicate codes in javascript.

Reply:
i'll come back to you regarding this part as i need more time to investigate them.
but, i think there is a reason why i have `layers` and `map_layers` on map edit page.